File: android/src/main/kotlin/net/niuxiaoer/flutter_gromore/view/FlutterGromoreBase.kt
package net.niuxiaoer.flutter_gromore.view

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

/**
 * 广告基类
 * 必须实现 initAd
 * postMessage 发送事件
 */
abstract class FlutterGromoreBase(messenger: BinaryMessenger, channelName: String) {

    private val methodChannel = MethodChannel(messenger, channelName)

    // 初始化广告
    abstract fun initAd()

    // 发送事件消息
    protected fun postMessage(method: String, arguments: Map<String, Any?>? = null) {
        methodChannel.invokeMethod(method, arguments)
    }
}

File: ios/Classes/view/FlutterGromoreBase.swift
//
//  FlutterGromoreBase.swift
//  flutter_gromore
//
//  Created by jlq on 2022/6/1.
//

import Foundation

protocol FlutterGromoreBase {
    var methodChannel: FlutterMethodChannel? { get set }
    /// 初始化广告
    func initAd()
}

extension FlutterGromoreBase {
        
    func initMethodChannel(channelName: String, messenger: FlutterBinaryMessenger) -> FlutterMethodChannel? {
        let methodChannel = FlutterMethodChannel(name: channelName, binaryMessenger: messenger)
        return methodChannel
    }
    
    func postMessage(_ method: String, arguments: Any? = nil) {
        methodChannel?.invokeMethod(method, arguments: arguments)
    }
}
